### PR TITLE
Order history

### LINKED ISSUE #38: my-account: order history
- block to be loaded in static `/static/my-account/order-history.html`
- table view of orders
- filter by `date, user, status, total $`
- search by `PO#, order#, eecol#, SKU, product name`
- relies on API endpoint

**Links**
- [related API issue](https://github.com/hlxsites/eecol-cloudflare-worker/issues/17)

### LINKED ISSUE #50: my-account: role management
- admin only?

### LINKED ISSUE #54: my-account: order history - detail page
- past order, detailed breakdown of:
  - order status (last updated time)
  - items -> `item-summary` -> no link to product
  - price -> `order-summary`
  - shipping details
  - customer details
  - order notes
- reorder button

### PR DESCRIPTION
**Pull request by maxakuru**
closes #54
closes #50 
closes #38

related pr: https://github.com/hlxsites/eecol-react/pull/17

Test URLs:
- Before: https://main--eecol--hlxsites.hlx.page/
- After: https://order-history--eecol--hlxsites.hlx.page/

### CODE REVIEW

File: types/Auth.d.ts [hunk 1]
```diff
@@ -10,13 +10,33 @@
...
+  | 'Magento_Sales::view_orders' | 'Magento_Sales::view_orders_sub' | 'Magento_NegotiableQuote::all' | 'Magento_NegotiableQuote::view_quotes'
+  | 'Magento_NegotiableQuote::manage' | 'Magento_NegotiableQuote::checkout' | 'Magento_NegotiableQuote::view_quotes_sub' | 'Magento_PurchaseOrder::all'
+  | 'Magento_PurchaseOrder:view_purchase_orders' | 'Magento_PurchaseOrder:view_purchase_orders_for_subordinates' | 'Magento_Company::credit'
+  | 'Magento_PurchaseOrder:view_purchase_orders_for_company' | 'Magento_PurchaseOrder:autoapprove_purchase_order' | 'Magento_Company::users_edit'
+  | 'Magento_PurchaseOrder:super_approve_purchase_order' | 'Magento_PurchaseOrder:view_approval_rules' | 'Magento_Company::users_view'
+  | 'Magento_PurchaseOrder:manage_approval_rules' | 'Magento_Company::view' | 'Magento_Company::view_account' | 'Magento_Company::roles_edit'
+  | 'Magento_Company::edit_account' | 'Magento_Company::view_address' | 'Magento_Company::edit_address' | 'Magento_Company::credit_history'
+  | 'Magento_Company::contacts' | 'Magento_Company::payment_information' | 'Magento_Company::shipping_information'
+  | 'Magento_Company::user_management' | 'Magento_Company::roles_view';
+
```
Comment (mbouxsein24) [inline types/Auth.d.ts:27]: Hopefully temporary?

**Comment by rsfierst** [comment]
How can this be tested given the bug that prevents logging in?

File: types/Auth.d.ts (hunk 1 above)
Comment (maxakuru) [reply types/Auth.d.ts:27]: these are defined by magento as base permissions IDs and we're using them as types.. so no, why would they be temporary?

**Comment by maxakuru** [comment]
> How can this be tested given the bug that prevents logging in?

I think locally for now

**Review by mbouxsein24 — APPROVED** [review]

**Pull request merged**